### PR TITLE
HOCS-3512 Key Information - Swapping Post to Email does not retain th…

### PR DESCRIPTION
**Pull request by agraham19**
…e field values

* Redefine correspondent fields using visibilityConditions
* Fix country list

### CODE REVIEW

File: server/routes/api/case.js [hunk 1]
```diff
@@ -22,12 +22,15 @@ router.post('/:caseId/stage/:stageId/allocate/team',
...
-    caseApiResponseMiddleware
+router.post([
+    '/:caseId/stage/:stageId/entity/:entity/:context/:action',
+    '/:caseId/stage/:stageId/entity/:entity/:context/:caseType/:action',
+    '/:caseId/stage/:stageId/entity/:entity/:action'],
+getFormForCase,
+fileMiddleware.any(),
+processMiddleware,
+validationMiddleware,
+caseApiResponseMiddleware
```
Comment (gary-newbolt) [inline server/routes/api/case.js:33]: Can you tidy up the formatting here as well please?

File: server/routes/case.js [hunk 2]
```diff
@@ -16,7 +16,13 @@ router.all(['/:caseId/stage/:stageId', '/:caseId/stage/:stageId/allocate'],
...
     getDocumentList,
     getCaseNotes);
-router.all(['/:caseId/stage/:stageId/entity/:entity/:context/:action', '/:caseId/stage/:stageId/entity/:entity/:action'], getFormForCase, hydrateFields);
+
+router.all(['/:caseId/stage/:stageId/entity/:entity/:context/:action',
+    '/:caseId/stage/:stageId/entity/:entity/:context/:caseType/:action',
+    '/:caseId/stage/:stageId/entity/:entity/:action'
+], getFormForCase, hydrateFields
+);
+
```
Comment (gary-newbolt) [inline server/routes/case.js:25]: Again... slight formatting fix. It just makes it easier to read for future FOI team

File: server/routes/api/case.js (hunk 1 above)
Comment (agraham19) [reply server/routes/api/case.js:33]: The linter keeps complaining when I try anything else and 'corrects' it to this format.

File: server/routes/case.js (hunk 2 above)
Comment (agraham19) [reply server/routes/case.js:25]: The linter keeps complaining when I try anything else and 'corrects' it to this format.

File: server/routes/api/case.js (hunk 1 above)
Comment (gary-newbolt) [reply server/routes/api/case.js:33]: no worries... leave it as it is then

File: server/routes/case.js (hunk 2 above)
Comment (gary-newbolt) [reply server/routes/case.js:25]: no worries... leave it as it is then

File: server/services/forms/schemas/decorators/form_decorations/schema-decorations.js [hunk 3]
```diff
@@ -33,39 +33,142 @@ module.exports = {
...
+                                    }
+                                ])
+                                .build()
+                        )
+                        .withField(
+                            Component('dropdown', 'Country')
+                                .withValidator('required')
+                                .withProp('label', 'Country')
+                                .withProp('choices', 'COUNTRIES_CURRENT')
+                                // .withProp('defaultOptions', true)
```
Comment (gary-newbolt) [inline server/services/forms/schemas/decorators/form_decorations/schema-decorations.js:123]: Sorry... just spotted a random commented out line  as well. Do you need it or can it be removed?

**Review by gary-newbolt — APPROVED** [review]
great thanks

**Pull request merged**